From 721f739d2861e89f0b47c7948750e443de4d52f4 Mon Sep 17 00:00:00 2001
From: Nicolo Giuliani <nicologiuliani6@studio.unibo.it>
Date: Sat, 19 Sep 2026 20:12:13 +0200
Subject: [PATCH 1/3] kernel: add kmod-ntb-epf and kmod-ntb-netdev

kmod-ntb-epf is the host side of a PCIe endpoint NTB link (ntb, ntb_transport,
ntb_hw_epf, ntb_netdev). kmod-ntb-netdev is the transport and Ethernet client
for the endpoint side, where ntb and the endpoint function are built in.
---
 package/kernel/linux/modules/other.mk | 34 +++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/package/kernel/linux/modules/other.mk b/package/kernel/linux/modules/other.mk
index ad7ac9a..90e0c8b 100644
--- a/package/kernel/linux/modules/other.mk
+++ b/package/kernel/linux/modules/other.mk
@@ -1231,3 +1231,37 @@ define KernelPackage/regulator-userspace-consumer/description
 endef
 
 $(eval $(call KernelPackage,regulator-userspace-consumer))
+
+define KernelPackage/ntb-epf
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=PCIe endpoint NTB host side (virtual Ethernet to an endpoint device)
+  DEPENDS:=@PCI_SUPPORT
+  KCONFIG:= \
+	CONFIG_NTB \
+	CONFIG_NTB_EPF \
+	CONFIG_NTB_TRANSPORT \
+	CONFIG_NTB_NETDEV
+  FILES:= \
+	$(LINUX_DIR)/drivers/ntb/ntb.ko \
+	$(LINUX_DIR)/drivers/ntb/ntb_transport.ko \
+	$(LINUX_DIR)/drivers/ntb/hw/epf/ntb_hw_epf.ko \
+	$(LINUX_DIR)/drivers/net/ntb_netdev.ko
+  AUTOLOAD:=$(call AutoProbe,ntb ntb_transport ntb_hw_epf ntb_netdev)
+endef
+
+$(eval $(call KernelPackage,ntb-epf))
+
+define KernelPackage/ntb-netdev
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=Ethernet over an NTB (ntb_transport + ntb_netdev)
+  DEPENDS:=@PCI_SUPPORT
+  KCONFIG:= \
+	CONFIG_NTB_TRANSPORT \
+	CONFIG_NTB_NETDEV
+  FILES:= \
+	$(LINUX_DIR)/drivers/ntb/ntb_transport.ko \
+	$(LINUX_DIR)/drivers/net/ntb_netdev.ko
+  AUTOLOAD:=$(call AutoProbe,ntb_transport ntb_netdev)
+endef
+
+$(eval $(call KernelPackage,ntb-netdev))
-- 
2.48.1

